add Classes по Заданию

psasha007 opened the pull request:
Привет!
Добавил по заданию классы для проверок:
1. Пакет "questionspage"
1.1. Класс "HomePageQuestionsAboutImportantTest" - Тестовые сценарии для раздела "Вопросы о важном"

2. Пакет "pageobject"
2.1. Класс "DialogPageAboutRent" - Класс pageObject на страницу "Про аренду"
2.2. Класс "DialogPageWhoIsTheScooterFor" - Класс pageObject на страницу  "Для кого самокат"
2.3. Класс "OrderByScooterTest" - Тестовые сценарии для раздела "Заказ самоката"

3. Пакет "additionaltes"
3.1 Класс "additionalTest" - Дополнительные тестовые сценарии
3.2. Класс "ErrorsForOrderFormFields" - Класс pageObject на страницу "Для кого самокат" для проверки сообщений

psasha007 committed:
add Classes по Заданию

pomegranate282922 commented on src/test/java/additionaltes/ErrorsForOrderFormFields.java:
```diff
@@ -0,0 +1,112 @@
+
+    // метод проверяет сообщение об ошибке "Введите корректный адрес"
+    public void verifyFieldAdress(String adress, String expected) throws InterruptedException {
+        sleep(500);
```
Такие слипы используют в самом крайнем случае! Во всех остальных - используй явные ожидания, привязанные к ожиданию конкретного события.

psasha007 replied:
Слипы поправил, добавил явные ожидания

pomegranate282922 commented on src/test/java/additionaltes/additionalTest.java:
```diff
@@ -0,0 +1,107 @@
+
+import static java.lang.Thread.sleep;
+
+public class additionalTest {
```
Классы именуем с заглавной буквы!

psasha007 replied:
Сделал с заглавной буквы.

pomegranate282922 commented on src/test/java/additionaltes/additionalTest.java:
```diff
@@ -0,0 +1,107 @@
+    @Test
+    // Проверить: если нажать на логотип «Самоката», попадёшь на главную страницу «Самоката».
+    public void verifyScooterLogoHomePageTest(){
+        driver.findElement(By.className("Header_LogoScooter__3lsAR")).click();
```
В тестах не должно быть локаторов! Они должны быть описаны в page object, а на уровне тестов должны быть вызовы методов из po ( содержащие шаги и проверки)

psasha007 replied:
перенес локаторы в page object

pomegranate282922 commented on src/test/java/additionaltes/additionalTest.java:
```diff
@@ -0,0 +1,107 @@
+        sleep(1500);
+        Assert.assertTrue(actual.isDisplayed());
+    }
+
```
Везде здесь реализуй тесты с помощью page object model!

psasha007 replied:
реализовал тесты с помощью page object mode.

pomegranate282922 commented on src/test/java/pageobject/DialogPageAboutRent.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    // Метод отмечаем чек-бокс black
+    public void setCheakBoxBlack() {
```
Check

psasha007 replied:
Поправил Cheak на Check

pomegranate282922 commented on src/test/java/questionspage/HomePageQuestionsAboutImportantTest.java:
```diff
@@ -0,0 +1,162 @@
+    WebDriver driver;
+    String baseUrl = "https://qa-scooter.praktikum-services.ru/";
+    @Before
+    public void startUp() {
```
Такие общие для всех тестов начала и концы можно вынести в BaseTest с соответствующими аннотациями и сделать наследование реальных классов с тестами от него. Таким образом ты уберёшь ненужное дублирование кода.

psasha007 replied:
попробовал вариант BaseTest

pomegranate282922 commented on src/test/java/questionspage/HomePageQuestionsAboutImportantTest.java:
```diff
@@ -0,0 +1,162 @@
+    public void isTextHowMuchDoesItCostAndHowToPay() throws InterruptedException {
+        //new WebDriverWait(driver, 3);
+        sleep(1000);
+        driver.findElement(By.cssSelector("#accordion__heading-0")).click();
```
Тоже нужно сделать с помощью page object model. Это было обязательным требованием!

psasha007 replied:
сделал с помощью page object model.

pomegranate282922 reviewed (commented):
Работа хорошая, но есть и необходимость доработок.
Реализуй везде тесты с помощью page object model ( перенеси локаторы на уровень page object)
Везде убери слипы - вместо них используй явные ожидания соответсвующих событий.
Поправь нейминг кое-где
Остальное - по желанию.

psasha007 committed:
Class additionalTest переименовали с заглавной буквы!

psasha007 committed:
1. Добавил решение с помощью page object model для HomePageQuest...
2. Убрал seep(), добавил new WebDriverWait(driver, 10).until...
3. Cheak поправил на Cheсk
4. Из тестов убрал локаторы, описал в page object
5. Переименовал класс additionalTest в AdditionalTest
6. Поправил нейминг

psasha007 committed:
1. Добавил пробный BaseTest
2. Мелкие правки

psasha007 committed:
1. Мелкие правки

pomegranate282922 commented on src/test/java/additionaltes/AdditionalTest.java:
```diff
@@ -0,0 +1,59 @@
+    // Проверить: если нажать на логотип «Самоката», попадёшь на главную страницу «Самоката».
+    public void verifyScooterLogoHomePageTest(){
+        ErrorsForOrderFormFields testUrl = new ErrorsForOrderFormFields(driver);
+        testUrl.isCheckUrlScooterLogoHomePage("https://qa-scooter.praktikum-services.ru/");
```
Так используй константу! Зачем ее выносили?

psasha007 replied:
Теперь использую константу BASE_URL

pomegranate282922 commented on src/test/java/additionaltes/ErrorsForOrderFormFields.java:
```diff
@@ -0,0 +1,171 @@
+
+        new WebDriverWait(driver, 10).until(ExpectedConditions.visibilityOfElementLocated(locatorImageNotFound));
+        WebElement actual = driver.findElement(locatorImageNotFound);
+        Assert.assertTrue("Expected image not found", actual.isDisplayed());
```
Сначала в классе перечисляем все поля, потом все методы

psasha007 replied:
В классах вынес в начало переменные, потом методы

pomegranate282922 commented on src/test/java/pageobject/DialogPageWhoIsTheScooterFor.java:
```diff
@@ -0,0 +1,108 @@
+        WebElement element = driver.findElement(locatorDownButton);
+        ((JavascriptExecutor)driver).executeScript("arguments[0].scrollIntoView();", element);
+
+        new WebDriverWait(driver, 10).until(ExpectedConditions.visibilityOfElementLocated(locatorDownButton));
```
10 лучше вынести как публичную константу и везде переиспользовать.

psasha007 replied:
Добавил класс Consts и константу TIME_OUT_IN_SECONDS

pomegranate282922 commented on src/test/java/pageobject/OrderByScooterTest.java:
```diff
@@ -0,0 +1,90 @@
+
+        dialogPageWhoIs.clickHomePageDownButton();
+        dialogPageWhoIs.setFieldsWhoIs(userName, surName, adress, nameMetro, numbetForCourier);
+        dialogPageAboutRent.setFieldsAboutRent(date, textForCourier, verifyText);
```
Я бы не делала такие огромные шаги. Лучше помельче. И проверку выделить в отдельный шаг.

psasha007 replied:
Сделал шаги помельче, и проверку выделил в отдельный шаг

pomegranate282922 commented on src/test/java/questionspage/HomePageQuestions.java:
```diff
@@ -0,0 +1,134 @@
+    private By isItPossible = By.cssSelector("#accordion__heading-3");
+    private By isItPossibleText =  By.cssSelector("#accordion__panel-3");
+    // Метод проверяет текст "Только начиная с завтрашнего дня. Но скоро станем расторопнее."
+    public void isCheckIsItPossibleToOrderAScooterRightForTodayText(String expected){
```
Слишком огромный шаг -вообще про все! Раздели на действие и проверку!

psasha007 replied:
Разделил шаги на действие и проверку

pomegranate282922 commented on src/test/java/questionspage/HomePageQuestionsAboutImportantTest.java:
```diff
@@ -0,0 +1,84 @@
+    // HowMuchDoesItCostAndHowToPay - Сколько это стоит? И как оплатить?
+    @Test
+    public void isTextHowMuchDoesItCostAndHowToPayTest() throws InterruptedException {
+        HomePageQuestions Questions = new HomePageQuestions(driver);
```
Переменные именуем со строчной буквы!

psasha007 replied:
Переменные переименовал со сточной буквы

pomegranate282922 commented on src/test/java/questionspage/HomePageQuestionsAboutImportantTest.java:
```diff
@@ -0,0 +1,84 @@
+    @Test
+    public void isTextILiveOutsideTheMoscowRingRoadWillYouBringItTest() throws InterruptedException {
+        HomePageQuestions Questions = new HomePageQuestions(driver);
+        Questions.isCheckILiveOutsideTheMoscowRingRoadWillYouBringIt("Да, обязательно. Всем самокатов! И Москве, и Московской области.");
```
Попрошу все же разбить этот шаг на несколько , чтобы тесты стали более понятными.

psasha007 replied:
постарался разбить на несколько шагов

pomegranate282922 reviewed (commented):
Замечания многие поправил - спасибо за работу! Чуть ещё попрошу доделать - небольшие вроде, но все вместе делают работу не до конца законченной.

psasha007 committed:
1. Теперь использую константу BASE_URL.
2. В классах вынес в начало переменные, потом методы.
3. Добавил класс Consts и константу TIME_OUT_IN_SECONDS.
4. Переменные переименовал со сточной буквы.
5. Разделил шаги на действие и проверку.
6. Сделал шаги помельче, и проверку выделил в отдельный шаг.

psasha007 committed:
1. Мелкие правки

pomegranate282922 reviewed (approved):
Отлично!
